Sources/Apollo/HTTPNetworkTransport.swift

```diff
@@ -69,31 +69,38 @@ public protocol HTTPNetworkTransportRetryDelegate: HTTPNetworkTransportDelegate
 
 /// A network transport that uses HTTP POST requests to send GraphQL operations to a server, and that uses `URLSession` as the networking implementation.
 public class HTTPNetworkTransport: NetworkTransport {
+  
   let url: URL
-  let session: URLSession
+  var session: URLSession
   let serializationFormat = JSONSerializationFormat.self
-  let useGETForQueries: Bool
   let delegate: HTTPNetworkTransportDelegate?
 
+  private let _useGETForQueries: Bool
+  private let _enableAutoPersistedQueries: Bool
+  private let _useHttpGetMethodForPersistedQueries: Bool
+  
   /// Creates a network transport with the specified server URL and session configuration.
   ///
   /// - Parameters:
   ///   - url: The URL of a GraphQL server to connect to.
   ///   - configuration: A session configuration used to configure the session. Defaults to `URLSessionConfiguration.default`.
-  ///   - sendOperationIdentifiers: Whether to send operation identifiers rather than full operation text, for use with servers that support query persistence. Defaults to false.
   ///   - useGETForQueries: If query operation should be sent using GET instead of POST. Defaults to false.
+  ///   - enableAutoPersistedQueries: Whether to send persistedQuery extension. QueryDocument will be absent at 1st request, retry with QueryDocument if server respond PersistedQueryNotFound or PersistedQueryNotSupport. Defaults to false.
+  ///   - useHttpGetMethodForPersistedQueries: Whether to send PersistedQuery supported request with HTTPGETMethod, retry with HTTPPOSTMethod if PersistedQuery not support/not found in server.
```

Comment:
Why does this need to be a separate parameter instead of respecting the user's preference in `useGETForQueries`?

Reply:
Well, `useGETForQueries` is applying to all query requests, including those with query document.
Query doucement can be very large depends on usage. 

And query string will be super long too.
For those Graphql server which restricts the length of url for security reason would return error of `exceed url length limit`. 
In fact, many of http server restricts url length to 2k by default.

So we need another `useHttpGetMethodForPersistedQueries` for those who wanna set limit to their server for security or other reason. To only apply Get for those APQs succeed Requests (which without query document).

Reply:
updated